src/Microsoft.AspNetCore.Mvc.DataAnnotations/Internal/DataAnnotationsMetadataProvider.cs

```diff
@@ -80,23 +88,59 @@ public void CreateDisplayMetadata(DisplayMetadataProviderContext context)
                 displayMetadata.DataTypeName = DataType.Html.ToString();
             }
 
+
+            var localizedType = context.Key.ContainerType ?? context.Key.ModelType;
+            var localizer = _stringLocalizerFactory?.Create(localizedType);
+
             // Description
             if (displayAttribute != null)
             {
-                displayMetadata.Description = () => displayAttribute.GetDescription();
-                displayMetadata.Placeholder = () => displayAttribute.GetPrompt();
+                if (localizer != null &&
+                    !string.IsNullOrEmpty(displayAttribute.Description) &&
+                    displayAttribute.ResourceType == null)
+                {
+                    displayMetadata.Description = () => localizer[displayAttribute.GetDescription()].Value;
```

Comment:
Deserves a comment that `GetDescription()` will return `Description` when `ResourceType == null`. Either that or use `Description` here.